internal/server/data/grant.go

```diff
@@ -106,8 +104,91 @@ func GetGrant(tx ReadTxn, opts GetGrantOptions) (*models.Grant, error) {
 	return (*models.Grant)(table), nil
 }
 
-func ListGrants(db GormTxn, p *Pagination, selectors ...SelectorFunc) ([]models.Grant, error) {
-	return list[models.Grant](db, p, selectors...)
+type ListGrantsOptions struct {
+	BySubject   uid.PolymorphicID
+	ByPrivilege string
+	ByResource  string
+
+	// IncludeInheritedFromGroups instructs ListGrants to include grants from
+	// groups where the user is a member. This option can only be used when
+	// BySubject is a non-zero userID.
+	IncludeInheritedFromGroups bool
+
+	// IncludeConnectorGrant instructs ListGrants to exclude grants where the
+	// CreatedBy is models.CreatedBySystem.
+	IncludeConnectorGrant bool
+
+	Pagination *Pagination
+}
+
+func ListGrants(tx ReadTxn, opts ListGrantsOptions) ([]models.Grant, error) {
+	table := grantsTable{}
+	query := querybuilder.New("SELECT")
+	query.B(columnsForSelect(table))
+	if opts.Pagination != nil {
+		query.B(", count(*) OVER()")
```

Comment:
This is a change to how we get the total count of results. It's the same approach I used for access keys, but different from what we had with gorm.